cas/src/main/java/org/springframework/security/cas/authentication/CasAuthenticationProvider.java

```diff
@@ -187,6 +188,16 @@ public void setAuthenticationUserDetailsService(
 		this.authenticationUserDetailsService = authenticationUserDetailsService;
 	}
 
+	/**
+	 * Sets the UserDetailsChecker to be used for checking the status of retrieved user details.
+	 * This allows customization of the UserDetailsChecker implementation.
+	 *
+	 * @param userDetailsChecker the UserDetailsChecker to be set
+	 */
+	public void setUserDetailsChecker(final UserDetailsChecker userDetailsChecker) {
+		this.userDetailsChecker = userDetailsChecker;
```

Comment:
Can you please verify that the `userDetailsChecker` is not null?

```suggestion
        Assert.notNull(userDetailsChecker, "userDetailsChecker cannot be null");
		this.userDetailsChecker = userDetailsChecker;
```